### PR TITLE
feat(test-runner): shorten long output paths

### PR DESCRIPTION
**Pull request by JoelEinbinder**
See comment in #10447﻿

### CODE REVIEW

**Comment by tymfear** [comment]
Fixes #10361

File: packages/playwright-test/src/util.ts [hunk 1]
```diff
@@ -144,6 +144,16 @@ export function sanitizeForFilePath(s: string) {
   return s.replace(/[\x00-\x2C\x2E-\x2F\x3A-\x40\x5B-\x60\x7B-\x7F]+/g, '-');
 }
 
+export function trimLongString(s: string, length = 100) {
+  if (s.length <= length)
+    return s;
+  const hash = calculateSha1(s);
+  const middle = `-${hash.substring(5)}-`;
```
Comment (mxschmitt) [inline packages/playwright-test/src/util.ts:151]: `[...hash].splice(0, 5)` maybe instead?

File: tests/playwright-test/test-output-dir.spec.ts [hunk 2]
```diff
@@ -429,6 +429,21 @@ test('should allow nonAscii characters in the output dir', async ({ runInlineTes
...
+      const { test } = pwt;
+      test.describe('this is a really long description that would be too long for a file path', () => {
+        test('and this is an even longer test name that just keeps going and going and we should shorten it', async ({}, testInfo) => {
+          console.log('\\n%%' + testInfo.outputDir);
+        });
+      });
+    `,
+  });
+  const outputDir = result.output.split('\n').filter(x => x.startsWith('%%'))[0].slice('%%'.length);
+  expect(outputDir).toBe(path.join(testInfo.outputDir, 'test-results', 'my-test-this-is-a-really-long-descripti-f9015f78f2853e47676ac0890dcea2f2b85-d-going-and-we-should-shorten-it'));
```
Comment (mxschmitt) [inline tests/playwright-test/test-output-dir.spec.ts:444]: this still contains the full hash, shouldn't it be only the last 5 chars of the sha1?

Comment (JoelEinbinder) [reply tests/playwright-test/test-output-dir.spec.ts]: Yep. Silly mistake.

**Review by pavelfeldman — APPROVED** [review]

**Review by mxschmitt — APPROVED** [review]

**Pull request merged**